src/ui/public/vis/editors/default/sidebar.js

```diff
@@ -18,6 +19,14 @@ uiModules
       $scope.$watch('vis.type', (visType) => {
         if (visType) {
           this.showData = visType.schemas.buckets || visType.schemas.metrics;
+          if (_.has(visType, 'editorConfig.optionTabs')) {
+            const activeTabs = visType.editorConfig.optionTabs.filter((tab) => {
+              return _.get(tab, 'active', false);
+            });
+            if (activeTabs.length > 0) {
+              this.section = activeTabs[0].name;
+            }
+          }
           this.section = this.section || (this.showData ? 'data' : 'options');
```

Comment:
what happens here at this line, when `showData===false`, but none of the options tabs were set to active? Shouldn't we default to the first option tab then iso hardcode options here?